StoreKitTests/StoreKitTests.swift

```diff
@@ -46,9 +55,221 @@ class StoreKitTests: XCTestCase {
             receivedOfferings = offerings
         }
         expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
-        
+
         expect(receivedError).to(beNil())
         expect(receivedOfferings).toNot(beNil())
         expect(receivedOfferings!.all).toNot(beEmpty())
     }
+
+    func testCanMakePurchase() throws {
+        configurePurchases()
+        purchaseMonthlyOffering()
+
+        waitUntilPurchaseGoesThrough()
+        let entitlements = purchasesDelegate.purchaserInfo?.entitlements
+        expect(entitlements?["premium"]?.isActive) == true
+    }
+
+    func testPurchaseMadeBeforeLogInIsRetainedAfter() {
+        configurePurchases()
+
+        var completionCalled = false
+        purchaseMonthlyOffering { [self] purchaserInfo, error in
+            expect(purchaserInfo?.entitlements.all.count) == 1
+            let entitlements = self.purchasesDelegate.purchaserInfo?.entitlements
+            expect(entitlements?["premium"]?.isActive) == true
+
+            let anonUserID = Purchases.shared.appUserID
+            let identifiedUserID = "\(#function)_\(anonUserID)_".replacingOccurrences(of: "RCAnonymous", with: "")
+
+            Purchases.shared.logIn(identifiedUserID) { identifiedPurchaserInfo, created, error in
+                expect(error).to(beNil())
+
+                expect(created).to(beTrue())
+                expect(identifiedPurchaserInfo?.entitlements["premium"]?.isActive) == true
+                completionCalled = true
+            }
+        }
+        expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
+    }
+
+    func testPurchaseMadeBeforeLogInWithExistingUserIsNotRetainedAfterUntilRestore() {
```

Comment:
naming is hard, okay? seriously, if these names aren't clear or you have suggestions for better names, I'm more than open to them

Reply:
updated!